src/TraceEvent/TraceLog.cs

```diff
@@ -543,88 +760,66 @@ private TraceLog()
         /// Functionality of TraceLog that does not depend on either remembering past EVENTS or require future
         /// knowledge (e.g. stacks of kernel events), will 'just work'.
         /// </summary>
-        private unsafe TraceLog(TraceEventSession session)
+        private unsafe TraceLog(TraceEventDispatcher source)
```

Comment:
I think we will want to do some real-time ETW testing here to ensure that nothing has regressed in that path.

Reply:
There doesn't seem to be any test case for `CreateFromTraceEventSession()`, i.e. the ETW one - how do you think it would be best to go around testing this (and the EventPipeSession factory too)?

Reply:
Good question - there aren't any tests for this.  We don't really have an environment where we can run such tests elevated in an automated fashion.

I suspect the best thing to do here is to write some tests that emit events while a real-time session of each type is live, and then confirm that those events come through.

The EventPipe one can be automated if we so desire, because it doesn't require admin privileges.  The ETW one will likely just need to remain adhoc.

Reply:
Alright, I'll implement the EventPipe session tests similar to what I've done in Sentry dotnet SDK - there should be no problem running them in CI.

Reply:
I've added the test. After much headscratching why it was always freezing without an error I've found out that's what happens when you try starting an EventPipe session on a DiagnosticClient connection to a process that is older than .NET core 3.0... 
Since I was trying to connect to the test process itself to test the session, I had to add another `targetframework` to the test project and run the new test conditionally only on the newer .NET. I hope that's alright.

Reply:
Yes, that's great.  I don't think the IPC channel exists on .NET Core 3.x, so I wouldn't expect it to work there.

Reply:
It does work since .NET core 3.0. It's only missing on .NET framework.

Also, since I've found out the .netcoreapp3.1 is already being installed in the existing azure pipelines CI workflow, I've adapted the targetframework from net6.0 to netcoreapp3.1 (already in use in `PerfViewCollect` project